### ISSUE TITLE
web: rollup executor invalid type-checking

### ISSUE THREAD

**Issue by Hotell** [nrwl/nx#9053]
## Current Behavior

@nrwl/web:rollup executor uses under the hood https://github.com/ezolenko/rollup-plugin-typescript2 TS plugin.

Now because this plugin doesn't work as expected https://github.com/ezolenko/rollup-plugin-typescript2/issues/298 one may/will ship invalid code to consumers.

## Expected Behavior

rollup executor `@nrwl/web:rollup` should properly run type-checking to be on par with raw `tsc`

### Environment

```
yarn : 1.22.10
  
  nx : 13.4.6
  @nrwl/angular : undefined
  @nrwl/cli : 13.4.6
  @nrwl/cypress : 13.4.6
  @nrwl/devkit : 13.4.6
  @nrwl/eslint-plugin-nx : 13.4.6
  @nrwl/express : undefined
  @nrwl/jest : 13.4.6
  @nrwl/linter : 13.4.6
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : 13.4.6
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.4.6
  @nrwl/web : 13.4.6
  @nrwl/workspace : 13.4.6
  @nrwl/storybook : 13.4.6
  @nrwl/gatsby : undefined
  typescript : 4.4.4
  rxjs : 6.6.7
```

**Comment by Hotell** [comment]
NOTE: I tried to use official [typescript rollup plugin](https://github.com/rollup/plugins/tree/master/packages/typescript) - it type-checks correctly so maybe I'd reconsider to use that one for rollup executor (would be great to have some visibility why a non official plugin is being used ATM) .

**Comment by ronnetzer** [comment]
Hi @Hotell, any new regarding this one?
We've noticed that rollup-plugin-typescript2 has some performance issues compared to the official plugin which is significantly faster.

**Comment by StoneYaoTreelab** [comment]
> The issue has been closed as fixed in that roll-up plugin repo, so executor deps need to be bumped. We went with our own implementation to unblock. Cheers

But even I migrate the nx to 15.0.0, the type check error still exist and the tsc type check is fine.

**Comment by jaysoo** [comment]
Fixed in [`95a3bbfa](https://github.com/nrwl/nx/commit/95a3bbfabff1ffcb4713e9998bbc9bd34118c150).
